sycl/source/detail/device_impl.cpp

```diff
@@ -435,6 +436,46 @@ std::string device_impl::getDeviceName() const {
   return MDeviceName;
 }
 
+/* On first call this function queries for device timestamp
+   along with host synchronized timestamp
+   and stores it in memeber varaible deviceTimePair.
+   Succive calls to this function would just retrieve the host timestamp ,
+   compute difference against the host timestamp in deviceTimePair
+   and calculate the device timestamp based on the difference.
+   deviceTimePair is refreshed with new device and host timestamp after a
+   certain interval (determined by timeTillRefresh) to account for clock skew
+   between host and device.
+*/
+
+uint64_t device_impl::getTime(){
+  constexpr uint64_t timeTillRefresh = 100e9;
```

Comment:
Could you please clarify if there is any idea behind this specific value `100e9`?

Reply:
I just chose an arbitrary number longer than a minute

Reply:
Is that correct that this refresh needed to deal with counter overflow in LevelZero. If so, we try to calculate how much time the counter would not overflow by taking current device time and "timestampValidBits".

Reply:
I forgot to add this to the documentation, but the purpose for `timeTillRefresh` is to account for clock drift between the device and host; where the clocks can fall out of phase over a long period of time, so we need to resynchronize them on regular intervals.

Reply:
As for the overflow problem in L0, I couldn't find a good way to deal with it, so I didn't